components/eam/src/physics/cam/micro_p3.F90

```diff
@@ -3154,7 +3154,7 @@ subroutine update_prognostic_ice(qc2qi_hetero_freeze_tend,qccol,qc2qr_ice_shed_t
    if (do_predict_nc) then
       nc = nc + (-nc_collect_tend-nc2ni_immers_freeze_tend)*dt
    endif
-   qr = qr + (-qrcol+qi2qr_melt_tend-qr2qi_immers_freeze_tend+qc2qr_ice_shed_tend)*dt
+   qr = qr + (-qrcol-qr2qi_immers_freeze_tend+qc2qr_ice_shed_tend)*dt
```

Comment:
Pinging these changes so they show up in the comments.  I'm going to remove the commits that make this bugfix.  Is it causing `p3_run_and_cmp` to fail and is holding up the PR.  Since they are orthogonal to the main thrust of this fix they can be removed.  But we should submit an issue that points to the bug.